### PR TITLE
90: user/role table migration and seeding

### PR DESCRIPTION
**Pull request by NickPhura**
- Remove previously added migration that re-created all of the legacy lucy-web tables.
  - Because invasivesbc doesn't care about them (yet/maybe never?), and because they are fairly out of date now that we have a new codes set/value setup.
- Add migration to do initial db setup
  - In this case just creating the schema.
- Add migration for the application_user and user_role tables (same as lucy-web database)
- Add seed for test users and their roles (same as lucy-web)

### CODE REVIEW

**Review by rstens — COMMENTED** [review]
Looks good

File: database/src/migrations/20200810113233_init.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function down(knex: Knex): Promise<void> {
+  await knex.raw(`
+    set schema '${DB_SCHEMA}';
+    set search_path = ${DB_SCHEMA},public;
+
+    DROP SCHEMA IF EXISTS ${DB_SCHEMA}.invasivesbc;
```
Comment (anissa-agahchen) [inline database/src/migrations/20200810113233_init.ts]: This is breaking the downgrade.  Just need to remove `.invasivesbc`
```suggestion
    DROP SCHEMA IF EXISTS ${DB_SCHEMA};
```

File: database/src/migrations/20200810113233_init.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`
+    set schema '${DB_SCHEMA}';
+    set search_path = ${DB_SCHEMA},public;
+
+    CREATE SCHEMA IF NOT EXISTS invasivesbc;
+
+    ALTER SCHEMA invasivesbc OWNER TO invasivebc;
```
Comment (anissa-agahchen) [inline database/src/migrations/20200810113233_init.ts]:
```suggestion
    ALTER SCHEMA ${DB_SCHEMA} OWNER TO invasivebc;
```

File: database/src/migrations/20200810113233_init.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`
+    set schema '${DB_SCHEMA}';
+    set search_path = ${DB_SCHEMA},public;
+
+    CREATE SCHEMA IF NOT EXISTS invasivesbc;
```
Comment (anissa-agahchen) [inline database/src/migrations/20200810113233_init.ts]:
```suggestion
    CREATE SCHEMA IF NOT EXISTS ${DB_SCHEMA};
```

**Review by anissa-agahchen — CHANGES_REQUESTED** [review]
@NickPhura Looks really good.  
Just found one small breaking change on the migration downgrade, and put a couple of suggestions.

File: database/src/migrations/20200810113233_init.ts (hunk 1 above)
Comment (NickPhura) [reply database/src/migrations/20200810113233_init.ts]: Nice catch!

Also, never seen this 'commit suggestion' option before, thats interesting

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**